portal_spa/src/components/forms/relying-party/child-components/AddHostnameFields.tsx

```diff
@@ -0,0 +1,52 @@
+import { Button } from "@/components/ui/button";
+import { FormMessage } from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import type {
+  FieldArrayWithId,
+  FieldErrors,
+  UseFormRegister,
+} from "react-hook-form";
+import type { RelyingPartyFormData } from "../validation-schema";
+
+type AddHostnameFieldsProps = {
+  field: FieldArrayWithId<RelyingPartyFormData, "hostnames">;
+  index: number;
+  register: UseFormRegister<RelyingPartyFormData>;
+  removeHostname: (index: number) => void;
+  errors: FieldErrors<RelyingPartyFormData>;
+};
+export default function AddHostnameFields(props: AddHostnameFieldsProps) {
+  const { field, index, register, removeHostname, errors } = props;
+  return (
+    <>
+      <div key={index} className="flex gap-2 items-start">
+        {field.id !== undefined && typeof field.id === "number" && (
```

Comment:
The second condition makes the first one redundant.

```suggestion
        {typeof field.id === "number" && (
```